contracts/liquidity_hub/pool-manager/src/tests/integration_tests.rs

```diff
@@ -2649,3 +2650,436 @@ mod locking_lp {
+
+                    match err {
+                        ContractError::AssetMismatch { .. } => {}
+                        _ => panic!("Wrong error type, should return ContractError::AssetMismatch"),
+                    }
+                },
+            )
+            .provide_liquidity(
+                creator.clone(),
+                "whale-uluna".to_string(),
+                None,
+                None,
+                vec![Coin {
+                    denom: "uwhale".to_string(),
+                    amount: Uint128::from(1_000_000u128),
+                }],
+                |result| {
+                    let err = result.unwrap_err().downcast::<ContractError>().unwrap();
+
+                    match err {
+                        ContractError::InvalidZeroAmount {} => {}
+                        _ => panic!(
+                            "Wrong error type, should return ContractError::InvalidZeroAmount"
+                        ),
+                    }
+                },
+            )
+            .provide_liquidity(
+                creator.clone(),
+                "whale-uluna".to_string(),
+                None,
+                None,
+                vec![Coin {
+                    denom: "uwhale".to_string(),
+                    amount: Uint128::from(1_000_000u128),
+                }],
+                |result| {
+                    let err = result.unwrap_err().downcast::<ContractError>().unwrap();
+
+                    // it was trying to provide single-side liquidity with no funds in the pool
+                    match err {
+                        ContractError::InvalidZeroAmount {} => {}
+                        _ => panic!(
+                            "Wrong error type, should return ContractError::InvalidZeroAmount"
+                        ),
+                    }
+                },
+            );
+
+        // let's provide liquidity with two assets
+        suite
+            .provide_liquidity(
+                creator.clone(),
+                "whale-uluna".to_string(),
+                None,
+                None,
+                vec![
+                    Coin {
+                        denom: "uwhale".to_string(),
+                        amount: Uint128::from(1_000_000u128),
+                    },
+                    Coin {
+                        denom: "uosmo".to_string(),
+                        amount: Uint128::from(1_000_000u128),
+                    },
+                ],
+                |result| {
+                    let err = result.unwrap_err().downcast::<ContractError>().unwrap();
+
+                    // it was trying to provide single-side liquidity with no funds in the pool
+                    match err {
+                        ContractError::AssetMismatch {} => {}
+                        _ => panic!("Wrong error type, should return ContractError::AssetMismatch"),
+                    }
+                },
+            )
+            .provide_liquidity(
```

Comment:
Seems duplicated

Reply:
kinda but not, the order or the assets is inverted but yeah